### ISSUE TITLE
Dealing with non-convergence issues

### ISSUE THREAD

**Issue by gragusa** [gragusa/MomentBasedEstimators.jl#7]
In application is common for nonlinear GMM problem to fail to converge to a minimum. In this case the status field of `GMMResult` will contain the flag from `MathProgBase`. We should decide what will happen (especially in iterated GMM) when an iteration fails to converge.

**Comment by sglyon** [comment]
Agreed. The paranoid way would be to immediately throw an error if we encounter anything but `:Optimal` from `MathProgBase`. That might be too strict though?

**Comment by gragusa** [comment]
Probably to extreme, but probably necessary. 

We could throw an error in the high level API, and printing a warning when using the low level API.
